src/main/java/company/evo/elasticsearch/script/PositionRecipScript.java

```diff
@@ -52,11 +53,19 @@ private PositionRecipScript(double m, double a, double b, double c, SearchLookup
     }
 
     @Override
-    public double execute() {
+    public double execute(ExplanationHolder explanation) {
         return m / (a * (Double) variables.get(GroupingMixupRescorer.POSITION_PARAMETER_NAME) + b) + c;
     }
 
-    public static class PositionRecipFactory implements Factory {
+    public static class PositionRecipFactory implements ScoreScript.Factory, ScriptFactory {
+
+        @Override
+        public boolean isResultDeterministic() {
+            // PositionRecipScript does not use deterministic APIs so that means that results
+            // are not cachable? @todo
```

Comment:
As I understand it is deterministic as it doesn't use any API like `random`.

Reply:
Y that was my understanding also. Lets remove the @todo comment then?